### ISSUE TITLE
Grammar-based SIE parsing

### ISSUE THREAD

**Issue by olleolleolle** [byrokrat/accounting#2]
After initial talk, I tried a Ruby programming approach to parsing SIE content.

https://gist.github.com/olleolleolle/cd29a9cb697f0f0b6f843bd24fab83d2 is a grammar and a runner with two ultra-simple words to test, but they're tested against the grammar file.

The third word is not matched by the grammar.

**Comment by hanneskod** [comment]
So after a few stumbling steps I have cloned your parser using `php-peg`:

https://github.com/byrokrat/tmp-sie-parser

Not sure I like the comment block syntax, but the library seems powerful enough and I cannot find another suitable php project...

The repo contains some additional files for compiling the parser and setting up a simple test environment. Once the matching strategy starts to form iteslf the code will move here. For now it's just a playground :smile:

**Comment by hanneskod** [comment]
https://github.com/scato/phpeg

**Comment by hanneskod** [comment]
Commit [c0e64a1](https://github.com/byrokrat/tmp-sie-parser/commit/c0e64a1a10b94d0a5890fc67628f6f06dc65237d) implements the same parser using phpeg, and looks a lot more promising..

The next step is probably to start implementing the various rules from section 5 in the [SIE spec](https://github.com/byrokrat/accounting/blob/master/docs/SIE_filformat_ver_4B_080930.pdf). And maybe define a core API and a generic testing strategy at this stage...

``` php
$someCleverFormat = (new SieParserFacade)->parse(file_get_contents($pathToSource));
```

...

**Comment by hanneskod** [comment]
Closed in favour of #3.
